### ISSUE TITLE
MOSES dataset: AttributeError: 'NoneType' object has no attribute 'edge_list'

### ISSUE THREAD

**Issue by scintiller** [DeepGraphLearning/torchdrug#14]
I tried following the tutorial of [GCPN](https://torchdrug.ai/docs/tutorials/generation.html#pretraining-and-generation-gcpn). But when I execute `model = models.RGCN(input_dim=dataset.node_feature_dim,
                    num_relation=dataset.num_bond_type,
                    hidden_dims=[256, 256, 256, 256], batch_norm=False)`, I got the following error:
```
Traceback (most recent call last):
  File "gcpn.py", line 13, in <module>
    num_relation=dataset.num_bond_type,
  File "/opt/conda/lib/python3.8/site-packages/torchdrug/data/dataset.py", line 168, in num_bond_type
    return len(self.bond_types)
  File "/opt/conda/lib/python3.8/site-packages/torchdrug/utils/decorator.py", line 21, in __get__
    result = self.func(obj)
  File "/opt/conda/lib/python3.8/site-packages/torchdrug/data/dataset.py", line 183, in bond_types
    bond_types.update(graph.edge_list[:, 2].tolist())
AttributeError: 'NoneType' object has no attribute 'edge_list'
```

I tried the dataset MOSES. It looks like in `torchdrug/data/dataset.py`, line 183, we have 
```
for graph in self.data:
       bond_types.update(graph.edge_list[:, 2].tolist())
```
and then the program complained about that `graph` is a `NoneType`, which may refer to `self.data` also has some problem. So I think there may be some bug in RGCN when interpreting the dataset. But I don't know how the inside works, could someone help solve this problem?

To reproduce this problem, I used nvidia docker:
```
nvidia-docker run -it --name=xxxxx  nvcr.io/nvidia/pytorch:21.06-py3 /bin/bash
conda install -c milagraph -c conda-forge torchdrug
```
and then follow the tutorial.

**Comment by KiddoZhu** [comment]
Hi! Since MOSES dataset is very large and we load it in a lazy fashion in TorchDrug. See #10 for how to fix this problem.

We will try to find a better interface for datasets, so that users don't need to explicitly deal with lazy loading.

**Comment by Oxer11** [comment]
Hi! This bug has been fixed in commit 6151fa9, where we load the whole dataset when calling the `bond_types` function. 

To fix the problem, you can download the [latest version](https://anaconda.org/milagraph/torchdrug/files?__cf_chl_jschl_tk__=pmd_PpqCdxwAnvDzp55wWjZsY8Ajnc6JaVPCox4WqaF_2pI-1629609878-0-gqNtZGzNAiWjcnBszQpR) of torchdrug from conda.
